lib/cpp/test/CMakeLists.txt

```diff
@@ -25,6 +25,10 @@ include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
 #Make sure gen-cpp files can be included
 include_directories("${CMAKE_CURRENT_BINARY_DIR}")
 
+if (MINGW)
+  add_definitions(-DBOOST_TEST_DYN_LINK)
+endif()
+
```

Comment:
You're right, it's not related to mingw strictly, it's related to the way you build Boost under MinGW. Your idea is fine, I was finding a way to detect if Boost requires dynamic linking automatically, but I didn't find any method. We could ignore this diff, if that's not possible.